primitives/src/witness.rs

```diff
@@ -185,37 +185,41 @@ impl Witness {
 
     /// Returns the last element in the witness, if any.
     #[inline]
-    pub fn last(&self) -> Option<&[u8]> {
-        if self.witness_elements == 0 {
-            None
-        } else {
-            self.nth(self.witness_elements - 1)
-        }
-    }
-
-    /// Returns the second-to-last element in the witness, if any.
-    #[inline]
-    pub fn second_to_last(&self) -> Option<&[u8]> {
```

Comment:
Good idea refactoring these out.  A little surprised this made it in to begin with.